src/serac/infrastructure/application_manager.hpp

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2019-2025, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#pragma once
+
+#include <string>
+#include <utility>
+#include <iostream>
+
+#include "mpi.h"
+
+namespace serac {
+
+/**
+ * @brief Destroy MPI, signal handling, logging, profiling, hypre, sundials, petsc, and slepc.
+ */
+void finalizer();
```

Comment:
I don't think we want to expose this to users. We want it to be called either by the RAII of ApplicationManager or via the catching of the signals.